libs/agent-runner.js

```diff
@@ -353,6 +440,22 @@ agent.prototype.runPushOperation = function runPushOperation(operation) {
         return;
       }
 
+      if (encryptionKey && commits.length) {
+        log.info('[ENCRYPTION] Encrypting data before it\'s committed...');
+
+        commits.forEach((entry) => {
+          operation.encrypt.fields.forEach((field) => {
+            if (!_.get(entry.data, field)) {
+              return;
+            }
+
+            entry.data[field] = Crypt.salt(encryptionKey).encrypt(entry.data[field]);
+          });
+        });
+
+        log.info('[ENCRYPTION] All data have been encrypted.');
```

Comment:
Grammar: 'All data **has** been encrypted.'

Reply:
@ibroom data is a plural. Datum is the singular of data.

Reply:
Fun! I love debating grammar.
From https://www.theguardian.com/news/datablog/2010/jul/16/data-plural-singular
> And the Guardian style guide expresses it like this:
> 
> Data takes a singular verb (like agenda), though strictly a plural; no one ever uses "agendum" or "datum"

So although you are right, data is used in the singular and plural in english therefore 'data has' (the singular) is correct as we are referring to a singular even though it technically is a plural.

A single data is often called a record. Data is made of records.

The same as agenda = plural and singular. I have never used 'agendum' but it is technically correct.

Reply:
@ibroom funnily enough the same page you shared says:

> Most style guides and dictionaries have come to accept the use of the noun data with either singular or plural verbs, and we hereby join the majority.

> As usage has evolved from the word's origin as the Latin plural of datum, singular verbs now are often used to refer to collections of information: Little data is available to support the conclusions.

> **Otherwise, generally continue to use the plural: Data are still being collected.**

I agree that no one uses datum, but I still find very common for people to use the plural and "piece of data" for the singular. I have no problem in changing this but I think it's very trivial, opinionated and minor.

Fun fact: in Italian we still distinct the two. Dato (datum) and Dati (data).